wandb/sdk/launch/_project_spec.py

```diff
@@ -110,6 +110,7 @@ def __init__(
         self.deps_type: Optional[str] = None
         self._runtime: Optional[str] = None
         self.run_id = run_id or generate_id()
+        self._queue_name: Optional[str] = None
```

Comment:
I feel like we need to do a better job distinguishing the queue's name by itself and the fully qualified queue URI of `entity/project/queue` or `entity/queue`.  And in this case, this is actually the URI, right?

Reply:
totally could be the URI, right now its just the name. I actually do have a question about this, which is how wandb bills users for run storage. Would sticking things in this config contribute (if minutely) to their costs? probably still worth it to make the URI, just wondering.... @nickpenaranda

Reply:
also using the URI here would make parsing on the backend more straightforward

Reply:
great point, IMO this should not be billed under run storage but included under some kind of flat fee associated with their license.... but that's a decision for another day.

Reply:
yes, after looking at the draft backend PR, totally agree having the URI here can simplify that a lot.  let's take these complexity-reducing Wins while we can

Reply:
Currently we have no way of going from `<entity>/<project>/<queue-name>` to the actual queue in the UI, whats the point? Especially if all queues are created in the `model-regsitry` project, the only thing thats sort of interesting is the `entity`, which we always know from the run anyway right?

Reply:
> the only thing thats sort of interesting is the entity, which we always know from the run anyway right?

this would be the entity that owns the queue, not the entity of the destination project.  Is that available on the run somewhere?

one thing that bothers me about our "queue URIs" is the fact that they can easily be mistaken for other resources... if i squint i could mistake it for any other project scoped resource.  It feels like overkill but I want something like `queue:0:<entity>/<project>/<queue_name>` and `queue:1:<entity>/<queue_name>` here

Reply:
@nickpenaranda Would the intent be to use this when the uri is in a position that a user could see it? Or should we store this even internally so the uri is clear for devs? I'm on board with adding identifiers to queue uris.

@gtarpenning If we ever allow runs from a queue in entity 1 to go to entity 2 just having the queue name will lead to some funky behavior, since I assume we'll be getting the queue from the config and the entity from the run. For example, the W&B global queue would break if we store the queue name only right now.

>Currently we have no way of going from <entity>/<project>/<queue-name> to the actual queue in the UI, whats the point?
You can query for the run queue by name on the project in our schema.

Reply:
yeah, there are other problems like the lack of a route like `wandb.ai/<entity>/<launch>/queue/<name>`, but I think we can add this and migrate to it without affecting the original route.  We'll probably want to redirect the other route to the new one.

Reply:
Thats a good point about the cross queue bit @KyleGoyette. And yes I am aware that we can query for the queue in mds. But again, the only unique advantage of including the entity/project would be for instances where run entity != queue entity, which currently we don't allow (?) 

To Nicks point, I agree the current queue URI scheme is opaque and could be made more user friendly. If we an align on what that URI would look like definitely, then it would definitely make sense to include that URI in the config in this PR.

Reply:
^ OoS for now

Reply:
We explicitly allow it for the W&B global queue, this would break queue observability for the global queue. Given that we have in the past made a cross entity queue, we may allow it again in the future. And the relative simplicity of including the queue's entity here, I don't see why we would not?

Reply:
Yep, totally, just want to lock down the format.